README.md

```diff
@@ -1,289 +1,590 @@
+   * [Provisioned Service](#provisioned-service)
+      * [Resource Type Schema](#resource-type-schema)
+      * [Example Resource](#example-resource)
+      * [Well-known Secret Entries](#well-known-secret-entries)
+   * [Application Projection](#application-projection)
+      * [Example Directory Structure](#example-directory-structure)
+   * [Service Binding](#service-binding)
+      * [Resource Type Schema](#resource-type-schema-1)
+      * [Example Resource](#example-resource-1)
+      * [Reconciler Implementation](#reconciler-implementation)
+   * [Extensions](#extensions)
+      * [Binding Secret Generation Strategies](#binding-secret-generation-strategies)
+         * [OLM Operator Descriptors](#olm-operator-descriptors)
+         * [Descriptor Examples](#descriptor-examples)
+         * [Non-OLM Operator and Resource Annotations](#non-olm-operator-and-resource-annotations)
+         * [Annotation Examples](#annotation-examples)
+      * [Mapping Existing Values to New Values](#mapping-existing-values-to-new-values)
+         * [Resource Type Schema](#resource-type-schema-2)
+         * [Example Resource](#example-resource-2)
+      * [Binding Values as Environment Variables](#binding-values-as-environment-variables)
+         * [Resource Type Schema](#resource-type-schema-3)
+         * [Example Resource](#example-resource-3)
+      * [Synthetic Provisioned Service](#synthetic-provisioned-service)
+         * [Resource Type Schema](#resource-type-schema-4)
+         * [Example Resource](#example-resource-4)
+
+<!-- Added by: bhale, at: Mon Jun  1 20:46:50 PDT 2020 -->
+
+<!--te-->
+
+## Notational Conventions
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
+
+The key words "unspecified", "undefined", and "implementation-defined" are to be interpreted as described in the [rationale for the C99 standard](http://www.open-std.org/jtc1/sc22/wg14/www/C99RationaleV5.10.pdf#page=18).
+
+An implementation is not compliant if it fails to satisfy one or more of the MUST, MUST NOT, REQUIRED, SHALL, or SHALL NOT requirements for the protocols it implements.  An implementation is compliant if it satisfies all the MUST, MUST NOT, REQUIRED, SHALL, and SHALL NOT requirements for the protocols it implements.
 
-Specification for binding services to runtime applications running in Kubernetes.  
+## Terminology definition
 
-## Conventions
+<dl>
+  <dt>Duck Type</dt>
+  <dd>Any type that meets the contract defined in a specification, without being an instance of a specific concrete type.  For example, for specification that requires a given key on <code>status</code>, any resource that has that key on its <code>status</code> regardless of its <code>kind</code> would be considered a duck type of the specification.</dd>
```

Comment:
Hi!
Would this be an example of a `Duck Type` ?
```
status:
  binding:
    metadata:
      name: mysql-binding
    secret:
      name: mysql-binding
```

Reply:
Correct, any resource of any kind that has that structure would implement this duck type.

We simplified the proposed bindable duck type to be:
```yaml
status:
  binding:
    name: mysql-binding
```

Reply:
@sbose78 as @scothis says, the type is simpler than I'd been suggesting previously, but the overall pattern is correct; [if it looks like a duck and quacks like a duck, it must be a duck](https://en.wikipedia.org/wiki/Duck_typing).  If there's a type that meets the contract (`.status.binding.name`) then it doesn't matter what it's concrete type is, it *is* that duck type.

Another way to look at this is that we're defining an interface that any type can implementation.  This works quite nicely for how Go developers view the world, because unlike something like Java, in Go interface implementations are not declared, they are implicitly implemented simply by adding the appropriate functions.

Reply:
Sounds good!